spec/lib/whitehall_importer/import_spec.rb

```diff
@@ -107,5 +113,28 @@
 
       expect(document.first_published_at).to eq(first_publish_date)
     end
+
+    it "integrity checks the current and live editions of the imported document" do
+      editions = [
+        build(:whitehall_export_edition),
+        build(:whitehall_export_edition, :published),
+      ]
+      described_class.call(build(:whitehall_export_document, editions: editions))
+
+      expect(WhitehallImporter::IntegrityChecker.new).to have_received(:valid?).twice
+    end
+
+    it "aborts with a list of problems if the integrity check fails" do
+      allow(WhitehallImporter::IntegrityChecker)
+        .to receive(:new)
+        .and_return(instance_double(
+                      WhitehallImporter::IntegrityChecker,
+                      valid?: false,
+                      problems: ["base path doesn't match"],
+                    ))
+
+      expect { described_class.call(build(:whitehall_export_document)) }
+        .to raise_error(WhitehallImporter::AbortImportError)
```

Comment:
Should we also assert that this includes the message since we typed it out above?